src/acceleration/impl/ParallelMatrixOperations.hpp

```diff
@@ -136,7 +136,6 @@ class ParallelMatrixOperations {
 
     PRECICE_ASSERT(_needCyclicComm);
     PRECICE_ASSERT(leftMatrix.cols() == q, leftMatrix.cols(), q);
-    PRECICE_ASSERT(leftMatrix.rows() == rightMatrix.cols(), leftMatrix.rows(), rightMatrix.cols());
```

Comment:
Removing this doesn't feel good.

Reply:
In the comment it says that ` For multiplication W_til * Z = J, p = r = n_global, q = m`, which is not the case now. I'm not so sure why this assertion is necessary for the operation unless these two variables are mixed used. Need to check.